packages/babel-traverse/src/scope/index.ts

```diff
@@ -392,9 +392,9 @@ class Scope {
   inited;
 
   bindings: { [name: string]: Binding };
-  references: { [name: string]: true };
+  references?: Set<string>;
   globals: { [name: string]: t.Identifier | t.JSXIdentifier };
-  uids: { [name: string]: boolean };
+  uids?: Set<string>;
```

Comment:
Can you add a comment saying that they are only defined in the program scope? Or maybe, we can simplify some methods by defining them in all scopes but using the same shared set.